### PR TITLE
Adding the query expansion article

### PR DESCRIPTION
**Pull request by TuanaCelik**

### CODE REVIEW

File: content/blog/query-expansion/index.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+However, retrieval is also in turn dependent on the query that it receives. There are multiple types of retrieval: keyword based, semantic search (embedding) based, hybrid, or even in some cases simply based on the results of a query to an API (for example, the results of websearch and so on). But at the end of the day, in the majority of cases, there’s a human behind a keyboard typing a query, and humans are not guaranteed to produce good quality queries for the results they intend to get.
+
+In this article, we’ll walk you through a very simple yet effective technique that allows us to make sure we are retrieving more of, and more relevant bits of context to a given query: query expansion.
+
+> TL;DR: Query expansion increases the number of results, so it increases recall (vs precision). In general, BM25 favors precision while embedding retrieval favors recall (See this [explanation by Nils Remiers](https://github.com/UKPLab/sentence-transformers/issues/22#issuecomment-529387645)). So, it makes sense to use BM25+query expansion to increase recall in cases where you want to rely on keyword search.
```
Comment (anakin87) [inline content/blog/query-expansion/index.md:22]:
```suggestion
> TL;DR: Query expansion increases the number of results, so it increases recall (vs precision). In general, BM25 favors precision while embedding retrieval favors recall (See this [explanation by Nils Reimers](https://github.com/UKPLab/sentence-transformers/issues/22#issuecomment-529387645)). So, it makes sense to use BM25+query expansion to increase recall in cases where you want to rely on keyword search.
```

File: content/blog/query-expansion/index.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    Document(content="The impact of climate change is evident in the melting of the polar ice caps."),
+    Document(content="Consequences of global warming include the rise in sea levels."),
+    Document(content="One of the effects of environmental changes is the change in weather patterns."),
+    Document(content="There is a global call to reduce the amount of air travel people take."),
+    Document(content="Air travel is one of the core contributors to climate change."),
+    Document(content="Expect warm climates in Turkey during the summer period."),
+]
+```
+
+When asking to retrieve the top 3 documents to the query "climate change” using the `InMemoryBM25Retriever` (so, we’re doing keyword search) here’s what we get as our top 3 candites:
```
Comment (anakin87) [inline content/blog/query-expansion/index.md:114]:
```suggestion
When asking to retrieve the top 3 documents to the query "climate change” using the `InMemoryBM25Retriever` (so, we’re doing keyword search) here’s what we get as our top 3 candidates:
```

**Review by anakin87 — APPROVED** [review]
Looks good.

(I left 2 small comments)

File: content/blog/query-expansion/index.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        llm = OpenAIGenerator(model = self.model)
+        self.pipeline = Pipeline()
+        self.pipeline.add_component(name="builder", instance=builder)
+        self.pipeline.add_component(name="llm", instance=llm)
+        self.pipeline.connect("builder", "llm")
+
+    @component.output_types(queries=List[str])
+    def run(self, query: str, number: int = 5):
+        result = self.pipeline.run({'builder': {'query': query, 'number': number}})
+        expanded_query = eval(result['llm']['replies'][0]) + [query]
```
Comment (anakin87) [inline content/blog/query-expansion/index.md:77]: see https://github.com/deepset-ai/haystack-cookbook/pull/107#discussion_r1715080030
